### DISCUSSION TITLE
Business API Authentication

### DISCUSSION THREAD

**Discussion by o-namacode** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#40] [Q&A]
How would I go about using this to implement the preliminary endpoint verification process before you set up the hooks? Rather, is it possible?
![image](https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/8162547/461fbd44-668b-4711-913c-37cdc2a22f64)

**Comment by gabrieldwight** [comment]
For verification, you will need to setup your callback URL to receive a get request from Meta and add your random string to verify the token. The verify token you set in meta must match with the verify token you set in the code and return a successful response to register your webhook. 

```C#
[HttpGet]
        public ActionResult<string> ConfigureWhatsAppMessageWebhook([FromQuery(Name = "hub.mode")] string hubMode,
                                                                    [FromQuery(Name = "hub.challenge")] int hubChallenge,
                                                                    [FromQuery(Name = "hub.verify_token")] string hubVerifyToken)
        {
            _logger.LogInformation("Results Returned from WhatsApp Server\n");
            _logger.LogInformation($"hub_mode={hubMode}\n");
            _logger.LogInformation($"hub_challenge={hubChallenge}\n");
            _logger.LogInformation($"hub_verify_token={hubVerifyToken}\n");

            if (!hubVerifyToken.Equals(VerifyToken))
            {
                return Forbid("VerifyToken doesn't match");
            }
            return Ok(hubChallenge);
        }
```

**Comment by divyaniverma** [comment]
There are a few procedures and criteria to acquire access to the WhatsApp Business API. Here's a general overview of the procedure:
Review the WhatsApp Business API Documentation
Apply as a WhatsApp Business Solution Provider (BSP)
Meet WhatsApp's Business Policy and Compliance Requirements
Set Up a WhatsApp Business Account
Integration and Development
Testing and Approval
Go Live and Start Using the WhatsApp Business API

Or you can simply connect with the whatsapp business solution provider such as msgclub who can provide you [whatsapp business API](https://www.msgclub.net/whatsapp-business.aspx) to connect your CRMs, Payment Portals, eCommerce Platforms via APIs. Easy to connect API for Developers or integration.
